addons/web/static/src/public/utils.js

```diff
@@ -109,3 +109,53 @@ export function makeButtonHandler(fct) {
         return result;
     };
 }
+
+/**
+ * Patches a "t-" entry of a dynamic content.
+ *
+ * @param {Object} dynamicContent
+ * @param {string} selector
+ * @param {string} t
+ * @param {any|function} replacement, if a function, takes the element and the
+ *     replaced's function output as parameters
+ */
+export function patchDynamicContentEntry(dynamicContent, selector, t, replacement) {
+    let forSelector = dynamicContent[selector];
+    if (!forSelector) {
+        forSelector = {};
+        dynamicContent[selector] = forSelector;
+    }
+    if (replacement === undefined) {
+        delete forSelector[t];
+    } else if (typeof replacement === "function") {
+        const oldFn = forSelector[t];
+        if (oldFn) {
+            forSelector[t] = (el, oldResult) => {
+                const result = oldResult || {};
+                Object.assign(result, oldFn(el, result));
+                Object.assign(result, replacement(el, result));
```

Comment:
Shouldn't you `.call(this)`?

Reply:
Also I wonder if a proper test "in real conditions" (particularly with t-on- listeners and t-att-class wouldn't be necessary)